query07.sql

```diff
@@ -3,4 +3,16 @@
 */
 
 -- Enter your SQL query here
-select ...
+with union_table as (
+  SELECT a.trip_id,a.duration,a.start_time::date,a.end_time::date,2019 as year
+  FROM indego_trips_2019_q2 a
+  UNION
+  SELECT b.trip_id,b.duration,b.start_time::date,b.end_time::date,2020 as year
+  FROM indego_trips_2020_q2 b)
+ 
+SELECT count(*),year
+FROM union_table
+WHERE EXTRACT(DAY FROM end_time::date)-EXTRACT(DAY FROM start_time::date) = 1
```

Comment:
Right track, but slightly under-counting. To really use this method, you would have to check for month boundaries as well (e.g. April 30 - May 1). The solution I used is [here](https://github.com/musa-509-fall-2021/assignment-01-intro-to-geographic-queries/blob/answers/query07.sql#L16-L25) (or a more compact equivalent solution [here](https://github.com/musa-509-fall-2021/assignment-01-intro-to-geographic-queries/blob/answers/query07.sql#L40-L42)). You could also [extract different parts from the date](https://www.postgresql.org/docs/9.1/functions-datetime.html#FUNCTIONS-DATETIME-EXTRACT), like `doy` (day of the year).